src/aws_encryption_sdk/internal/formatting/deserialize.py

```diff
@@ -61,109 +61,214 @@ def validate_header(header, header_auth, raw_header, data_key):
+    :return: Loaded encrypted data keys
+    :rtype: set of :class:`EncryptedDataKey`
+    """
+    (encrypted_data_key_count,) = unpack_values('>H', stream)
     encrypted_data_keys = set([])
     for _ in range(encrypted_data_key_count):
-        (key_provider_length,) = unpack_values('>H', tee_stream)
+        (key_provider_length,) = unpack_values('>H', stream)
         (key_provider_identifier,) = unpack_values(
             '>{}s'.format(key_provider_length),
-            tee_stream
+            stream
         )
-        (key_provider_information_length,) = unpack_values('>H', tee_stream)
+        (key_provider_information_length,) = unpack_values('>H', stream)
         (key_provider_information,) = unpack_values(
             '>{}s'.format(key_provider_information_length),
-            tee_stream
+            stream
         )
-        (encrypted_data_key_length,) = unpack_values('>H', tee_stream)
-        encrypted_data_key = tee_stream.read(encrypted_data_key_length)
+        (encrypted_data_key_length,) = unpack_values('>H', stream)
+        encrypted_data_key = stream.read(encrypted_data_key_length)
         encrypted_data_keys.add(EncryptedDataKey(
             key_provider=MasterKeyInfo(
                 provider_id=to_str(key_provider_identifier),
                 key_info=key_provider_information
             ),
             encrypted_data_key=encrypted_data_key
         ))
-    header['encrypted_data_keys'] = encrypted_data_keys
+    return encrypted_data_keys
 
-    (content_type_id,) = unpack_values('>B', tee_stream)
+
+def _verified_content_type_from_id(content_type_id):
+    # type: (int) -> ContentType
+    """Load a message :class:`ContentType` for the specified content type ID.
+
+    :param int content_type_id: Content type ID
+    :return: Message content type
+    :rtype: ContentType
+    :raises UnknownIdentityError: if unknown content type ID is received
+    """
     try:
-        content_type = ContentType(content_type_id)
+        return ContentType(content_type_id)
     except ValueError as error:
         raise UnknownIdentityError(
             'Unknown content type {}'.format(content_type_id),
             error
         )
-    header['content_type'] = content_type
 
-    (content_aad_length,) = unpack_values('>I', tee_stream)
+
+def _verified_content_aad_length(content_aad_length):
+    # type: (int) -> int
+    """Verify that content aad length is ``0``.
+
+    :param int content_aad_length: Content aad length to verify
+    :return: ``0``
+    :rtype: int
+    :raises SerializationError: if ``content_aad_length`` is not ``0``
+    """
     if content_aad_length != 0:
         raise SerializationError(
             'Content AAD length field is currently unused, its value must be always 0'
         )
-    header['content_aad_length'] = 0
 
-    (iv_length,) = unpack_values('>B', tee_stream)
-    if iv_length != alg.iv_len:
+    return 0
+
+
+def _verified_iv_length(iv_length, algorithm):
```

Comment:
Probably makes sense to rename algorithm -> algorithm_suite. Same for any other parameters where this has changed.

Reply:
Fair point. Updated within the changed inner functions. I'm going to leave the rest as-is for now, though. At some point I would like to go through the codebase and do a more complete transition, but I think this is not the time to do that.